cmd/faucet/faucet.go

```diff
@@ -141,31 +176,111 @@ func main() {
 		log.Crit("Failed to render the faucet template", "err", err)
 	}
 	// Load and parse the genesis block requested by the user
-	blob, err := ioutil.ReadFile(*genesisFlag)
-	if err != nil {
-		log.Crit("Failed to read genesis block contents", "genesis", *genesisFlag, "err", err)
-	}
-	genesis := new(core.Genesis)
-	if err = json.Unmarshal(blob, genesis); err != nil {
-		log.Crit("Failed to parse genesis block json", "err", err)
+	var genesis *core.Genesis
+	var enodes []*discv5.Node
+	var blob []byte
+
+	if *foundationFlag {
+		genesis = core.DefaultGenesisBlock()
```

Comment:
Extract method/lambda in here. Code duplication in here is enormous.
Something like:
```
func setConfig(newGenesis core.Genesis, bootnodes []string, newNetFlag string) {
    genesis = newGenesis
    if *bootFlag == "" {
			*bootFlag = strings.Join(bootnodes, ",")
		}
		if *netFlag == 0 {
			*netFlag = newNetFlag
		}
}
```
Maybe put those values to a map[